src/Control/Monad/Identity/Trans.purs

```diff
@@ -0,0 +1,112 @@
+module Control.Monad.Identity.Trans where
+
+import Prelude
+
+import Control.Alt (class Alt, (<|>))
+import Control.Alternative (class Alternative, empty)
+import Control.Monad.Cont.Class (class MonadCont, callCC)
+import Control.Monad.Error.Class (class MonadThrow, class MonadError, catchError, throwError)
+import Control.Monad.Reader.Class (class MonadAsk, class MonadReader, ask, local)
+import Control.Monad.Rec.Class (class MonadRec, tailRecM)
+import Control.Monad.State.Class (class MonadState, state)
+import Control.Monad.Trans.Class (class MonadTrans, lift)
+import Control.Monad.Writer.Class (class MonadWriter, class MonadTell, pass, listen, tell)
+import Control.MonadPlus (class MonadPlus)
+import Control.MonadZero (class MonadZero)
+import Control.Plus (class Plus)
+import Data.Eq (class Eq1)
+import Data.Foldable (class Foldable, foldl, foldr, foldMap)
+import Data.Traversable (class Traversable, traverse, sequence)
+import Data.Newtype (class Newtype)
+import Data.Ord (class Ord1)
+import Effect.Class (class MonadEffect, liftEffect)
+
+-- | The `IdentityT` monad transformer.
+-- |
+-- | This monad acts like a placeholder for functions that take a monad
+-- | transformer as an argument, similar to `identity` for functions and
+-- | `Identity` for monads.
+newtype IdentityT m a = IdentityT (m a)
+
+-- | Run a computation in the `IdentityT` monad.
+runIdentityT :: forall m a. IdentityT m a -> m a
+runIdentityT (IdentityT ma) = ma
+
+-- | Change the result type of a `IdentityT` monad action.
+mapIdentityT :: forall m1 m2 a b. (m1 a -> m2 b) -> IdentityT m1 a -> IdentityT m2 b
+mapIdentityT f (IdentityT m) = IdentityT (f m)
+
+derive instance eqIdentityT :: (Eq1 m, Eq a) => Eq (IdentityT m a)
+derive instance ordIdentityT :: (Ord1 m, Ord a) => Ord (IdentityT m a)
+derive instance eq1IdentityT :: (Eq1 m) => Eq1 (IdentityT m)
+derive instance ord1IdentityT :: (Ord1 m) => Ord1 (IdentityT m)
+derive instance newtypeIdentityT :: Newtype (IdentityT m a) _
+
+instance functorIdentityT :: Functor m => Functor (IdentityT m) where
+  map f (IdentityT m) = IdentityT (map f m)
+
+instance applyIdentityT :: Apply m => Apply (IdentityT m) where
+  apply (IdentityT mf) (IdentityT ma) = IdentityT (apply mf ma)
+
+instance applicativeIdentityT :: Applicative m => Applicative (IdentityT m) where
+  pure a = IdentityT (pure a)
+
+instance altIdentityT :: (Functor m, Alt m) => Alt (IdentityT m) where
+  alt (IdentityT x) (IdentityT y) = IdentityT (x <|> y)
+
+instance plusIdentityT :: (Monad m, Plus m) => Plus (IdentityT m) where
```

Comment:
I think this one only needs `Functor m` too?

Reply:
Yeah. Fixed.

Reply:
Does it really need Functor? For context, see this image of our type class hierarchy. Seems like `Plus` should be enough:
https://github.com/JordanMartinez/purescript-jordans-reference/blob/latestRelease/41-Ecosystem/Type-Classes/Type-Class-Relationships.svg

Reply:
Yes, good point! since Plus m already implies Alt m, which implies Functor m.